lib/models/user.ex

```diff
@@ -0,0 +1,48 @@
+defmodule SlackCoder.Models.User do
+  use SlackCoder.Web, :model
+
+  schema "users" do
+    field :slack, :string
+    field :github, :string
+    field :avatar_url, :string
+    field :html_url, :string
+    field :name, :string
+    field :monitors, StringList
+
+    embeds_one :config, SlackCoder.Models.User.Config
+
+    timestamps
+  end
+
+  @required_fields ~w(slack github config)
+  @optional_fields ~w(avatar_url html_url name config monitors)
+
+  @doc """
+  Creates a changeset based on the `model` and `params`.
+
+  If no params are provided, an invalid changeset is returned
+  with no validation performed.
+  """
+  def changeset(model, params \\ :empty) do
+    model
+    |> cast(params, @required_fields, @optional_fields)
```

Comment:
@mtyeh411 Fields not included in this `@required_fields`/`@optional_fields` list are not changeable, basically the params cannot affect the model for the missing fields. This way you can create multiple `changeset` functions which serve different use cases. For example, you could add an `admin_changeset` function which would have access to changing say a `:role` field if we had one. Then, elsewhere in your code only call the function which the user has access to call. It makes the code cleaner and more straightforward to read rather than relying on something like strong params to whitelist the fields correctly in the controller or relying on something internally to the model to do something similar.